aws/resource_aws_wafv2_ip_set.go

```diff
@@ -0,0 +1,255 @@
+				}
+				id := idParts[0]
+				name := idParts[1]
+				scope := idParts[2]
+				d.SetId(id)
+				d.Set("name", name)
+				d.Set("scope", scope)
+				return []*schema.ResourceData{d}, nil
+			},
+		},
+
+		Schema: map[string]*schema.Schema{
+			"addresses": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				MaxItems: 50,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					o, n := d.GetChange("addresses")
+					oldAddresses := o.(*schema.Set).List()
+					newAddresses := n.(*schema.Set).List()
+					if len(oldAddresses) == len(newAddresses) {
+						for _, ov := range oldAddresses {
+							hasAddress := false
+							for _, nv := range newAddresses {
+								// isIpv6CidrsEquals works for both IPv4 and IPv6
+								if isIpv6CidrsEquals(ov.(string), nv.(string)) {
+									hasAddress = true
+									break
+								}
+							}
+							if !hasAddress {
+								return false
+							}
+						}
+						return true
+					}
+					return false
+				},
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(1, 256),
+			},
+			"ip_address_version": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					wafv2.IPAddressVersionIpv4,
+					wafv2.IPAddressVersionIpv6,
+				}, false),
+			},
+			"lock_token": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringLenBetween(1, 128),
+			},
+			"scope": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					wafv2.ScopeCloudfront,
+					wafv2.ScopeRegional,
+				}, false),
+			},
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsWafv2IPSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).wafv2conn
+	params := &wafv2.CreateIPSetInput{
+		Addresses:        aws.StringSlice([]string{}),
+		IPAddressVersion: aws.String(d.Get("ip_address_version").(string)),
+		Name:             aws.String(d.Get("name").(string)),
+		Scope:            aws.String(d.Get("scope").(string)),
+	}
+
+	if v, ok := d.GetOk("addresses"); ok && v.(*schema.Set).Len() > 0 {
+		params.Addresses = expandStringSet(v.(*schema.Set))
+	}
+
+	if v, ok := d.GetOk("description"); ok {
+		params.Description = aws.String(v.(string))
+	}
+
+	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
+		params.Tags = keyvaluetags.New(v).IgnoreAws().Wafv2Tags()
+	}
+
+	resp, err := conn.CreateIPSet(params)
+
+	if err != nil {
+		return fmt.Errorf("Error creating WAFv2 IPSet: %s", err)
+	}
+
+	d.SetId(*resp.Summary.Id)
```

Comment:
we should use the AWS SDK's conversion function here as well around `resp.Summary.Id`